assets/tests/hooks/useDetour.test.ts

```diff
@@ -508,18 +508,60 @@ describe("useDetour", () => {
   })
 
   describe("stops", () => {
-    test("`stops` is initially populated with the stops from the original route shape", () => {
-      const stops = stopFactory.buildList(4)
+    test("`stops` is initially populated with the stops from the original route shape with a `missed` field added", () => {
+      const stop1 = stopFactory.build()
+      const stop2 = stopFactory.build()
+
+      const { result } = renderHook(() =>
+        useDetour(
+          originalRouteFactory.build({
+            shape: shapeFactory.build({ stops: [stop1, stop2] }),
+          })
+        )
+      )
+
+      const expectedStops = [
+        { ...stop1, missed: false },
+        { ...stop2, missed: false },
+      ]
+
+      expect(result.current.stops).toStrictEqual(expectedStops)
+    })
+
+    test("when the detour is finished, `stops` excludes any missed stops", async () => {
+      const stop1 = stopFactory.build()
+      const stop2 = stopFactory.build()
+      const stop3 = stopFactory.build()
+      const stop4 = stopFactory.build()
+
+      jest
+        .mocked(fetchFinishedDetour)
+        .mockResolvedValue(
+          finishedDetourFactory.build({ missedStops: [stop2, stop3] })
+        )
 
       const { result } = renderHook(() =>
         useDetour(
           originalRouteFactory.build({
-            shape: shapeFactory.build({ stops }),
+            shape: shapeFactory.build({ stops: [stop1, stop2, stop3, stop4] }),
           })
         )
       )
+      act(() => result.current.addConnectionPoint?.(shapePointFactory.build()))
+      act(() => result.current.addConnectionPoint?.(shapePointFactory.build()))
+
+      await waitFor(() => {
+        expect(result.current.missedStops).not.toHaveLength(0)
+      })
+
+      const expectedStops = [
+        { ...stop1, missed: false },
+        { ...stop2, missed: true },
+        { ...stop3, missed: true },
+        { ...stop4, missed: false },
+      ]
 
-      expect(result.current.stops).toBe(stops)
+      expect(result.current.stops).toStrictEqual(expectedStops)
     })
```

Comment:
question: is this assertion in the `waitFor` important?
It seems like you could instead do

```suggestion
      const expectedStops = [
        { ...stop1, missed: false },
        { ...stop2, missed: true },
        { ...stop3, missed: true },
        { ...stop4, missed: false },
      ]

      await waitFor(() => {
        expect(result.current.stops).toStrictEqual(expectedStops)
      })      
    })
```